cmd/describe/addon/cmd.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addon
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	errors "github.com/zgalor/weberr"
+
+	"github.com/openshift/moactl/pkg/logging"
+	"github.com/openshift/moactl/pkg/ocm"
+	rprtr "github.com/openshift/moactl/pkg/reporter"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "addon [ID|NAME]",
+	Aliases: []string{"add-on"},
+	Short:   "Show details of an add-on",
+	Long:    "Show details of an add-on",
+	Example: `  # Describe an add-on named "codeready-workspaces"
+  moactl describe addon codeready-workspaces`,
+	Run: run,
+}
+
+func run(_ *cobra.Command, argv []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Check command line arguments:
+	if len(argv) != 1 {
+		reporter.Errorf(
+			"Expected exactly one command line argument or flag containing the identifier of the add-on",
+		)
+		os.Exit(1)
+	}
+	addOnID := argv[0]
+
+	// Create the client for the OCM API:
+	ocmConnection, err := ocm.NewConnection().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmConnection.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Get the client for the OCM collection of add-ons:
+	addOnsCollection := ocmConnection.ClustersMgmt().V1().Addons()
+
+	// Try to find the add-on:
+	reporter.Debugf("Loading add-on '%s'", addOnID)
+	addOn, err := ocm.GetAddOn(addOnsCollection, addOnID)
+	if err != nil {
+		err = fmt.Errorf("Failed to get add-on '%s': %s", addOnID, err)
+		if errors.GetType(err) == errors.NotFound {
```

Comment:
you will have a duplicate error message here

Reply:
Good catch, thanks